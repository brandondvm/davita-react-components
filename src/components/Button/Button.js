import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ disabled, size, label, fill, type='button', onClick }) => (
  <button
    onClick={ onClick }
    className={ `btn ${fill} ${size}` }
    disabled={ !!disabled }
    type={ type }
  >
    { label }
  </button>
);

Button.propTypes = {
  disabled: PropTypes.bool,
  size: PropTypes.string,
  label: PropTypes.string,
  fill: PropTypes.string,
  onClick: PropTypes.func,
};

export default Button;
