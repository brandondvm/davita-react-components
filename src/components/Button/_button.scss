.btn {
  background-color: $blue;
  color: $white;
  font-family: $bold-font;
  font-size: 16px;
  height: 48px;
  width: 100%;

  &:focus {
    outline:none;

    &:active {
      outline:none;
    }
  }

  &:disabled {
    background-color: $palegray;
    border: none;
    color: $darkgray;
    cursor: not-allowed;
    filter: alpha(opacity=100);
    opacity: 1;
  }

  &.is-filled_white {
    background-color: $white;
    border-radius: 3px;
    border: 2px solid $blue;
    color: $blue;
  }

  &.not-filled {
    background-color: $white;
    color: $outerSpace;
    font-family: $bold-font;
  }
}
