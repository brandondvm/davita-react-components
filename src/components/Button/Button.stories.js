import React from 'react';
import Button from './Button';

export default {
  title: 'Button',
  component: Button,
};

const Template = (args) => <div className="storybook-wrapper"><Button {...args} /></div>;

export const Default = Template.bind({});
Default.args = {
  label: 'Default'
};

export const Disabled = Template.bind({});
Disabled.args = {
  label: 'Disabled',
  disabled: true
};
