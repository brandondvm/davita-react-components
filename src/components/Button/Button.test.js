import enzyme from 'enzyme';
import React from 'react';

import Button from './Button';

const mockProps = {
  disabled: false,
  size: 'small',
  label: 'Small button label',
  fill: 'blue',
  onClick: jest.fn(),
};

function setup(passedProps = {}) {
  const updatedProps = {
    ...mockProps,
    ...passedProps,
  };

  return enzyme.shallow(<Button { ...updatedProps } />);
}

describe('Button Component', () => {
  test('Component renders correctly when disabled prop is set to true', () => {
    const wrapper = setup({ disabled: true });

    expect(wrapper.get(0)).toMatchSnapshot();
  });

  test('Component renders correctly when disabled prop is set to false', () => {
    const wrapper = setup();

    expect(wrapper.get(0)).toMatchSnapshot();
  });

  test('Component registers click events', () => {
    const wrapper = setup();

    wrapper.find('button').simulate('click');
    expect(mockProps.onClick).toHaveBeenCalledTimes(1);
  });
});
