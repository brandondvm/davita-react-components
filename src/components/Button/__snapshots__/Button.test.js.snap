// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Component renders correctly when disabled prop is set to false 1`] = `
<button
  className="btn blue small"
  disabled={false}
  onClick={[MockFunction]}
  type="button"
>
  Small button label
</button>
`;

exports[`Button Component Component renders correctly when disabled prop is set to true 1`] = `
<button
  className="btn blue small"
  disabled={true}
  onClick={[MockFunction]}
  type="button"
>
  Small button label
</button>
`;
